## <center>Практическое задание к лекции 5.

**1. Известно, что генеральная совокупность распределена нормально, со средним квадратическим отклонением, $\displaystyle\sigma=16$. Найти доверительный интервал для оценки математического ожидания с надежностью $0.95$, если выборочная средняя $M=80$, а объем выборки $n=256$.**

Поскольку нам известно среднее квадратическое отклонение генеральной совокупности $\sigma=16$, то для расчета $95\%$-го доверительного интервала, будем использовать $Z$-критерий. И границы доверительного интервала расчитаем по формуле:  
$\displaystyle\bar X\pm z_{\alpha/2}\cdot\frac{\sigma}{\sqrt n}$, где:  
$\bar X=M=80$-среднее значение выборки    
$z_{\alpha/2}$ табличное значение $Z$-критерия, и для $95\%$-го доверительного интервала $z_{\alpha/2}=1,96$  
$\sigma=16$-среднее квадратическое отклонение генеральной совокупности  
$n=256$-объем выборки  
Таким образом $95\%$-й доверительный интервал расчитаем как: $\left[80-1.96\cdot\frac{16}{\sqrt 256};80+1.96\cdot\frac{16}{\sqrt 256}\right]$ 

In [1]:
left=80-1.96*16/256**(1/2)
right=80+1.96*16/256**(1/2)
print(f'95%-й доверительный интервал для оценки мат. ожидания генеральной совокупности: [{left};{right}].')

95%-й доверительный интервал для оценки мат. ожидания генеральной совокупности: [78.04;81.96].


---  
#### Ответ:  
$\left[78.04;81.96\right]$  

---

**2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.**

Поскольку из условия задачи нам ничего не известно ни о мат. ожидании генеральной совокупности ни о среднем квадратическом отклонении для генеральной совокупности, то для расчета $95\%$-го доверительного интервала будем использовать $t$-критерий,  и формулу:  
$\displaystyle\bar X\pm t_{\alpha/2}\cdot\frac{\sigma_{несмещ}}{\sqrt{n}}$, где  
$\sigma_{несмещ}$-несмещённое среднее квадратическое отклонение выборки,  
$n$-размер выборки  
Воспользуемся встроенными функциями библиотеки numpy:

In [5]:
import numpy as np

In [8]:
arr=np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
print(f'Среднее выборочное: {np.mean(arr): .2f},\n'
      f'Размер выборки n={len(arr)},\n'
      f'Среднее квадратическое отклонение по выборке(несмещенное): {np.std(arr, ddof=1): .2f}.'
     )

Среднее выборочное:  6.59,
Размер выборки n=10,
Среднее квадратическое отклонение по выборке(несмещенное):  0.45.


Для вычисления $t$-критерия воспользуемся библиотекой scipy

In [4]:
import scipy.stats as stats

In [21]:
def t_from_table(confidens, len_array):
    alpha=(1-confidens)
    return stats.t.ppf(1-alpha/2, len_array-1)
print(f'Табличное значение t-критерия для 95%-го доверительного интервала данной выборки: {t_from_table(0.95, len(arr)): .3f}')

Табличное значение t-критерия для 95%-го доверительного интервала данной выборки:  2.262


Итоговая функция для расчета доверительного интервала:

In [24]:
def confidens_int(arr, confidens):
    return round(np.mean(arr)-t_from_table(confidens,len(arr))*np.std(arr, ddof=1)/len(arr)**0.5,3), \
           round(np.mean(arr)+t_from_table(confidens,len(arr))*np.std(arr, ddof=1)/len(arr)**0.5,3)

print(f'95%-й доверительный интервал для истинного значения Х: {confidens_int(arr, 0.95)}.')

95%-й доверительный интервал для истинного значения Х: (6.268, 6.912).


---  
#### Ответ:  
$\left[6.268;6.912\right]$  

---

**3. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с $\alpha=0.05$ проверить эту гипотезу, если в выборке из $n=100$ шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв.мм.**  

Сформулируем нулевую гипотезу:  
$H0: \mu=\mu_0=17$, где $\mu$ и $\mu_0$ средние выборки и генеральной совокупности и выборки, соответственно.  
Альтернативная гипотеза:  
$H1: \mu>\mu_0$  
Поскольку нам известны и среднее и дисперсия генеральной совокупности, то для проверки гипотезы будем использовать $z$-критерий.  
$z_н=\frac{\bar X-\mu_0}{\sigma/\sqrt n}$, где  
$\bar X=\mu=17.5$ среднее по выборке,  
$\sigma=\sqrt 4=2$ среднее квадратическое отклонение генеральной совокупности,  
$n=100$ размер выборки.  

In [2]:
z=(17.5-17)/0.2
print(f'Наблюдаемое значение z-критерия: {z: .4f}')

Наблюдаемое значение z-критерия:  2.5000


Найдем табличное значение $z$-критерия, для $\alpha=0.05$

In [6]:
zt=stats.norm.ppf(0.95)
print(f'Табличное значение z-критерия: {zt: .4f}')

Табличное значение z-критерия:  1.6449


Сравним полученные значения: 

In [7]:
z<zt

False

Поскольку $z_н>z_t$, **отвергаем** нулевую гипотезу на уровне значимости $\alpha=0.05$

---  
#### Ответ:  
**Отвергаем** нулевую гипотезу $H0: \mu_0=17$ на уровне значимости $\alpha=0.05$  

---

**4. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?**  

Из условия задачи следует что нужно проводить проверку следующих гипотез:  
$H0: \mu=\mu_0=200$, где $\mu$ и $\mu_0$ средние выборки и генеральной совокупности и выборки, соответственно.  
Альтернативная гипотеза:  
$H1: \mu\neq\mu_0$  
при этом будем использовать **двусторонний** критерий c уровнем значимости $\alpha=1-0.99=0.01$  
Поскольку нам не известны ни среднее, ни дисперсия генеральной совокупности, то для проверки гипотезы будем использовать $t$-критерий.    

In [9]:
cookies=np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

Расчитаем наблюдаемое значение t-критерия $t_н$

In [11]:
tn=(np.mean(cookies)-200)/(np.std(cookies,ddof=1)/len(cookies))
print(f'Наблюдаемое значение t-критерия: {tn: .4f}')

Наблюдаемое значение t-критерия: -3.3682


Найдем табличное значение $t$-критерия $t_t$, для $\alpha/2=0.01/2=0.005$

In [21]:
tt=stats.t.ppf(0.995,len(cookies)-1)
print(f'Табличное значение t-критерия: {tt: .4f}')

Табличное значение t-критерия:  3.2498


Сравним полученные значения (поскольку у нас двусторонний критерий, то используем абсолютное значение полученного t-критерия):

In [23]:
np.abs(tn)<tt

False

Поскольку $|t_н|>t_t$, **отвергаем** нулевую гипотезу на уровне значимости $\alpha=0.001$

---  
#### Ответ:  
**Отвергаем** нулевую гипотезу $H0: \mu_0=200$ на уровне значимости $\alpha=0.001$  

---